### PR TITLE
Updates Optimistic Blue and adds “light” variant

### PR DESCRIPTION
**Pull request by jessicamarcus**
Related to https://github.com/CityOfBoston/design-team-workflow/issues/947

### CODE REVIEW

File: stylesheets/variables/_colors.styl
Comment (fionawhim) [inline stylesheets/variables/_colors.styl]: Can you add comments about when to use the light one? Is it background colors and text on dark backgrounds?

Comment (jessicamarcus) [reply stylesheets/variables/_colors.styl]: Totally can. 👍

**Review by fionawhim — APPROVED** [review]

**Review by fionawhim — APPROVED** [review]
Code approved, but don't check in w/o product / design approval, of course.

File: stylesheets/variables/_colors.styl
Comment (fionawhim) [inline stylesheets/variables/_colors.styl:19]: This is all very good. I'm glad you went with a complete renaming rather than having e.g. "optimistic-blue" and "optimistic-blue-light"

You *may* want to add a historical note comment that says which ones are the same hex #s as the original colors.

File: stylesheets/variables/_colors.styl
Comment (fionawhim) [inline stylesheets/variables/_colors.styl:34]: loving these meaningful names!

File: stylesheets/base/global/_text.styl
Comment (fionawhim) [inline stylesheets/base/global/_text.styl]: Should be changed to `#000000` since `--b` means "black" (instead of `--cb`)

File: stylesheets/components/form/_select.styl
Comment (fionawhim) [inline stylesheets/components/form/_select.styl:146]: 👍

File: stylesheets/components/form/_textbox.styl
Comment (fionawhim) [inline stylesheets/components/form/_textbox.styl:38]: hehe good tweak

File: stylesheets/variables/_colors.styl
Comment (jessicamarcus) [reply stylesheets/variables/_colors.styl:34]: 😊

File: stylesheets/base/global/_text.styl
Comment (jessicamarcus) [reply stylesheets/base/global/_text.styl]: I can do that - I think I did this because `$black` didn’t seem to be defined anywhere.

File: stylesheets/variables/_colors.styl
Comment (jessicamarcus) [reply stylesheets/variables/_colors.styl:19]: Thanks! I wanted to remove confusion, and this seemed to be a way to go. I’ll definitely add a comment to that effect; good suggestion.

**Comment by fionawhim** [comment]
Need to look into these cases:

 - [ ] Hub link colors (this is one snapshot of several): https://percy.io/CityOfBoston/patterns/builds/1385180/view/90996081/1280?mode=diff&browser=firefox&snapshot=90996081
 - [ ] Hub nav background: https://percy.io/CityOfBoston/patterns/builds/1385180/view/90996142/375?mode=diff&browser=firefox&snapshot=90996142
 - [ ] Freedom Red on Charles Blue: https://percy.io/CityOfBoston/patterns/builds/1385180/view/90995927/1280?mode=diff&browser=firefox&snapshot=90995927
 - [ ] Lost 311 yellow in footer: https://percy.io/CityOfBoston/patterns/builds/1385180/view/90996011/1280?mode=diff&browser=firefox&snapshot=90996011

**Comment by jessicamarcus** [comment]
legacy.css uses Fleet variables, so I added back `$optimistic-blue` and `$freedom-red` vars (pointing at their `-dark` variants) to catch those cases. We may want to do a rule-by-rule review to assign `-light` or `-dark` depending on case, but that file is 10k+ lines long with 60-something instances each for the two old variables... In a future version of Fleet, I'd recommend we create sets of color combination mixins along the lines of

```scss
@mixin colors-dark-bg {
  color: $white;
  background-color: $charles-blue;    
  a {
    color: $freedom-red-light;
    &:hover {
      color: currentColor;
    }
  }
}
``` 
and try to avoid specifying color in child components as much as possible...

**Comment by fionawhim** [comment]
Oooooh, I like that mixin a lot.

For legacy.css, can you put the compatibility color definitions at the top of it rather than in the general CSS to keep us from using them in clean code moving forward?

**Comment by jessicamarcus** [comment]
😁

re: color vars - I've been sort of intimidated by the thought of editing legacy.css, but moving them there makes way more sense!

File: legacy/public.styl
Comment (fionawhim) [inline legacy/public.styl:11]: 👍

Comment (fionawhim) [reply legacy/public.styl:11]: Clarify this based on our in-person convo.

**Review by fionawhim — APPROVED** [review]
LGTM, pending approval for Hub updates from Reilly.

**Review by fionawhim — APPROVED** [review]
Great work! Thanks for muddling through all of this.

**Comment by fionawhim** [comment]
GTG on Percy. Looks like your change reverted all but one of the Hub changes (which comes from the freedom red darkening)

**Pull request merged**
